connector/src/main/scala/com/microsoft/kusto/spark/datasource/KustoReader.scala

```diff
@@ -44,9 +45,11 @@ private[kusto] case class KustoReadRequest(sparkSession: SparkSession,
 
 private[kusto] case class KustoReadOptions(readMode: Option[ReadMode] = None,
                                            shouldCompressOnExport: Boolean = true,
-                                           exportSplitLimitMb: Long = 1024)
+                                           exportSplitLimitMb: Long = 1024,
+                                           exportQueryOnlyOnce: Boolean = false)
 private[kusto] object KustoReader {
   private val myName = this.getClass.getSimpleName
+  private val cachedQueryPaths = new ConcurrentHashMap[String, Seq[String]]().asScala
```

Comment:
the map should take into account also the KustoCoordinates and authenticaiton - as we do in "KustoRelation.Equals" method